packages/shared/src/utils/cookies.ts

```diff
@@ -43,3 +44,66 @@ export function isSecureEnvironment(headerHost?: string | string[]) {
 
   return true;
 }
+
+const decodeBase64URL = (value: string): string => {
+  try {
+    // atob is present in all browsers and nodejs >= 16
+    // but if it is not it will throw a ReferenceError in which case we can try to use Buffer
+    // replace are here to convert the Base64-URL into Base64 which is what atob supports
+    // replace with //g regex acts like replaceAll
+    return atob(value.replace(/[-]/g, '+').replace(/[_]/g, '/'));
+  } catch (e) {
+    if (e instanceof ReferenceError) {
+      // running on nodejs < 16
+      // Buffer supports Base64-URL transparently
+      return Buffer.from(value, 'base64').toString('utf-8');
```

Comment:
We've had issue in the past with Buffer not being available on cloudflare workers If I remember correctly, can we test this out before getting this merged in.

Reply:
That's why it falls back to `atob` it's taken from gotrue-js

Reply:
https://developers.cloudflare.com/workers/runtime-apis/web-standards/#base64-utility-methods

Reply:
Tested and confirmed that it works both in cludflare workers as well as vercel edge functions 👍